src/Components/Subscriptions/Form/index.js

```diff
@@ -0,0 +1,157 @@
+import styles from './form.module.css';
+import { useState, useEffect } from 'react';
+import { SubscriptionEditedModal, ErrorEdit } from '../Modals/index';
+
+function Form({ subscriptionId, onClose, onModalClose, updateTable }) {
+  const [valueClass, setValueClass] = useState('');
+  const [valueMember, setValueMember] = useState('');
+  const [valueDate, setValueDate] = useState('');
+  const [classes, setClasses] = useState([]);
+  const [members, setMembers] = useState([]);
+  const [dates, setDates] = useState([]);
+  const [showModal, setShowModal] = useState(false);
+  const [showServerError, setShowServerError] = useState(false);
+
+  useEffect(() => {
+    fetchClasses();
+    fetchMembers();
+    fetchSubscription();
+  }, []);
+
+  const fetchClasses = async () => {
+    try {
+      const response = await fetch(`${process.env.REACT_APP_API_URL}/api/class/search`);
+      const data = await response.json();
+      setClasses(data.data);
+    } catch (error) {
+      throw new Error(error);
+    }
+  };
+
+  const fetchMembers = async () => {
+    try {
+      const response = await fetch(`${process.env.REACT_APP_API_URL}/api/member`);
+      const data = await response.json();
+      setMembers(data.data);
+    } catch (error) {
+      throw new Error(error);
+    }
+  };
+
+  const fetchSubscription = async () => {
+    try {
+      const response = await fetch(`${process.env.REACT_APP_API_URL}/api/subscription`);
+      const data = await response.json();
+      setDates(data.data);
+    } catch (error) {
+      throw new Error(error);
+    }
+  };
+
+  function submitForm(event) {
+    event.preventDefault();
+    const updatedFields = {};
+
```

Comment:
we can add a validation, if there are not any changes, there is no need to make the request, and maybe, if we have time, we can add a modal saying that it didn't update because there are not changes